src/app/content/components/Page/PageComponent.tsx

```diff
@@ -47,6 +47,19 @@ export default class PageComponent extends Component<PagePropTypes> {
 
       transformContent(parsedContent, parsedContent.body, this.props.intl);
 
+      if (this.props.lockNavigation) {
+        const links = parsedContent.body.querySelectorAll('a[href]');
+
+        for (const link of links) {
+          const {origin, pathname} = new URL(link.getAttribute('href'), assertWindow().location.href);
+          const loc = assertWindow().location;
```

Comment:
this code needs to work in pre-rendering, so you can't use window.location, you need to use the navigation state path. you already have access to it at `props.currentPath`